### PR TITLE
#184581188 A User Should be able to list all Items 

### PR DESCRIPTION
**Pull request by wilbrord2**
#### What does this PR do?
This PR is for listing All products from Stock to the users

#### Description of Task to be completed?
-  creating routes for listing user items
- documenting routes in swagger
- test the endpoint

#### How should this be manually tested?
- Clone this repo git clone https://github.com/atlp-rwanda/e-comm-team-amigos-bn
- Switch your working directory cd e-comm-team-amigos-bn
- Create an .env file and update it with .env-example
- Install all the dependencies by running npm install
- Then run the project by typing npm run dev
- then run npm run dev
- open postman or other testing tools you have
- create a User with role of vendor or normal
- login with the user you created
- test end point `http://localhost:4000/product/getAllItems`

#### Endpoints to be tested:
- /user/login
- /user/otp
- /product/getAllItems

#### Any background context you want to provide?
- Have Node Js installed in your local machine
- postman (optional)

#### What are the relevant pivotal tracker/Trello stories?

Story ID: [#184581188](https://www.pivotaltracker.com/story/show/184581188)

#### Screenshots (if appropriate)

 Screenshot (137)

Screenshot (140)

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -0,0 +1 @@
+src/
```
Comment (Habinezajanvier) [inline .eslintignore]: Please add an empty line

**Review by Habinezajanvier — CHANGES_REQUESTED** [review]

File: src/controllers/product.controller.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+          Message: "List of all Products in our collections",
+          listProduct,
+        })
+        .status(200);
+    }
+  }
+  catch (error) {
+    res.status(401).json({
+      Message: error.message,
+    });
```
Comment (Habinezajanvier) [inline src/controllers/product.controller.js]: Please don't send message in errors

**Review by Habinezajanvier — CHANGES_REQUESTED** [review]

**Review by ericndungutse — APPROVED** [review]
LGTM

**Review by AbdulKhaliq59 — APPROVED** [review]

File: src/config.js [hunk 3]
```diff
@@ -11,12 +11,12 @@ module.exports = {
...
   },
   test: {
-    use_env_variable: '',
-    username: process.env.DB_USER,
-    password: process.env.DB_PASSWORD,
-    database: process.env.TEST_DB_NAME,
-    host: process.env.DB_HOST,
-    port: process.env.DB_PORT,
+    use_env_variable: 'DATABASE_URL',
+    // username: process.env.DB_USER,
```
Comment (nshutijonathan) [inline src/config.js]: get rid of commented out codes

File: src/config.js [hunk 4]
```diff
@@ -11,12 +11,12 @@ module.exports = {
...
+    // password: process.env.DB_PASSWORD,
+    // database: process.env.TEST_DB_NAME,
```
Comment (nshutijonathan) [inline src/config.js]: get rid of commented out codes

File: src/controllers/product.controller.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+        .json({
+          Status: "OK",
+          Message: "List of all Products in our collections",
+          listProduct,
+        })
+        .status(200);
+    }
+  } catch (error) {
+    res.status(401).json({
+      message: error,
```
Comment (nshutijonathan) [inline src/controllers/product.controller.js]: I would go with `message:`error.message`

File: src/tests/product.test.js [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+import models from "../database/models";
+import sinon from "sinon";
+import { v4 as uuid } from "uuid";
+import bcrypt from "bcryptjs";
+
+chai.use(chaiHttp);
+let user;
+// describe("Get all products", () => {
+//   before(async () => {
+//     await models.sequelize.sync({ force: true });
```
Comment (nshutijonathan) [inline src/tests/product.test.js]: remove commented out codes

**Review by nshutijonathan — CHANGES_REQUESTED** [review]
Check some comments

File: src/controllers/product.controller.js [hunk 7]
```diff
@@ -0,0 +1,38 @@
+import Joi from "joi";
+import models from "../database/models";
+export const addProduct = async (req, res) => {
+  try {
+    const product = await models.Product.create(req.body);
+    res.status(201).json(product);
+  } catch (error) {
+    res.status(400).json({ message: error.message });
```
Comment (Emile-SHUMBUSHO) [inline src/controllers/product.controller.js:8]: using `error: error.message` can be more descriptive and helpful  than` message: error.message`

**Review by Emile-SHUMBUSHO — CHANGES_REQUESTED** [review]

**Review by Emile-SHUMBUSHO — APPROVED** [review]

File: src/controllers/product.controller.js [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+};
+
+export const getAllProduct = async (req, res) => {
+  try {
+    const listProduct = await models.Product.findAll();
+    if (listProduct.length <= 0) {
+      res.status(404).json({
+        Status: "Not Found",
+        Message: "There is no product in Stock",
+      });
```
Comment (Habinezajanvier) [inline src/controllers/product.controller.js]: Please anything that is not sucess, I mean status code of 2XX, send `error: your error message` instead `message: error message`

**Review by Habinezajanvier — CHANGES_REQUESTED** [review]

**Review by nshutijonathan — APPROVED** [review]
LGTM

**Review by Habinezajanvier — APPROVED** [review]
👍

**Pull request merged**